android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java

```diff
@@ -46,15 +47,26 @@ public void onDataChange(DataSnapshot dataSnapshot) {
         if (values == null) {
             // TODO init database.
         } else if (values instanceof Map) {
+            int position = 0;
+            Iterator<String> iterator;
+
             Map<String, Map<String, Object>> data = (Map<String, Map<String, Object>>) values;
+            iterator = data.keySet().iterator();
             Log.i(TAG, "" + data);
             logs.clear();
-            for (Map<String, Object> entry : data.values()) {
+            for (Map<String, Object> entry: data.values()) {
```

Comment:
Consider
> for (Map<String, Object> entry : data.entrySet()) {
    String entryId = entry.getKey();
    Object moodEntry = entry.getValue();
    logs.add(new LogEntry(
        entryId,
        new Date((Long) entry.get("timestamp")),
        ((Long) entry.get("moodLevel")).intValue(),
         "" +entry.get("comment")));
}


This way you can fill in logs in one loop and avoid unnecessary complexity with iterators.

Reply:
Okay I'll try it.